src/js/actions/search/stock.js

```diff
@@ -0,0 +1,123 @@
+ * to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense,
+ * and/or sell copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ *
+ */
+
+define(function (require, exports) {
+    "use strict";
+
+    var Immutable = require("immutable");
+
+    var events = require("js/events"),
+        locks = require("js/locks"),
+        adapter = require("adapter"),
+        search = require("js/stores/search"),
+        nls = require("js/util/nls");
+
+    var STOCK_URL = search.SEARCH_STOCK_URL;
+    
+    /**
+     * Executes Adobe Stock search by opening a URL with the search term
+     *
+     * @return {Promise}
+     */
+    var _confirmStockSearch = function (searchTerm) {
+        var searchURL = STOCK_URL + searchTerm;
+        return adapter.openURLInDefaultBrowser(searchURL);
+    };
+    _confirmStockSearch.action = {
+        reads: [],
+        writes: []
+    };
+
+    /**
+     * Find SVG class for Adobe Stock
+     * If this needs to vary based on the item, use category list as parameter 
+     * (see getSVGCallback type in search store)
+     * 
+     * @return {string}
+    */
+    var _getSVGClass = function () {
+        return "libraries-stock";
+    };
+
+    /**
+     * Make list for Stock options, currently empty
+     * 
+     * @private
+     * @return {Immutable.List.<object>}
+    */
+    var _adobeStockSearchOptions = function () {
+        return Immutable.List();
+    };
+
+    /**
+     * Creates an object with properties to populate the search bar 
+     * datalist when there are no options present. 
+     * 
+     * @private
+     * @return {object}
+    */
+    var _createNoOptionsDefault = function () {
+        return {
+            "noOptionsString": nls.localize("strings.SEARCH.NO_OPTIONS_STOCK"),
+            "noOptionsExecute": null
+        };
+    };
+
+    /**
+     * Register Adobe Stock info for search
+     */
+    var registerAdobeStock = function () {
+        var adobeStockPayload = {
+            "type": "ADOBE_STOCK",
+            "getOptions": _adobeStockSearchOptions.bind(this),
+            "filters": Immutable.List.of("ADOBE_STOCK"),
+            "handleExecute": _confirmStockSearch.bind(this),
```

Comment:
It looks like you're just using the standard execution function to handle default options. If that works then that's fine. Maybe there doesn't need to be a special execution function just for default options?